c2corg_ui/static/js/editing/waypointediting.js

```diff
@@ -0,0 +1,242 @@
+goog.provide('app.WaypointEditingController');
+
+goog.require('app');
+goog.require('app.DocumentEditingController');
+goog.require('app.Document');
+
+
+/**
+ * @param {!angular.Scope} $scope Scope.
+ * @param {angular.JQLite} $element Element.
+ * @param {angular.Attributes} $attrs Attributes.
+ * @param {angular.$http} $http
+ * @param {Object} $uibModal modal from angular bootstrap.
+ * @param {angular.$compile} $compile Angular compile service.
+ * @param {app.Lang} appLang Lang service.
+ * @param {app.Authentication} appAuthentication
+ * @param {ngeo.Location} ngeoLocation ngeo Location service.
+ * @param {app.Alerts} appAlerts
+ * @param {app.Api} appApi Api service.
+ * @param {string} authUrl Base URL of the authentication page.
+ * @param {app.Document} appDocument
+ * @param {app.Url} appUrl URL service.
+ * @constructor
+ * @extends {app.DocumentEditingController}
+ * @ngInject
+ */
+app.WaypointEditingController = function($scope, $element, $attrs, $http,
+  $uibModal, $compile, appLang, appAuthentication, ngeoLocation, appAlerts,
+  appApi, authUrl, appDocument, appUrl, imageUrl) {
+
+  goog.base(this, $scope, $element, $attrs, $http, $uibModal, $compile,
+    appLang, appAuthentication, ngeoLocation, appAlerts, appApi, authUrl,
+    appDocument, appUrl, imageUrl);
+
+};
+
+goog.inherits(app.WaypointEditingController, app.DocumentEditingController);
+
+
+/**
+ * @param {appx.Document} doc Document attributes.
+ * @return {number}
+ * @override
+ */
+app.WaypointEditingController.prototype.presetQuality = function(doc) {
+  let score = 0;
+  const Type = doc['waypoint_type'];
```

Comment:
no uppercase, please

Reply:
``type`` is probably a reserved keyword. Use ``waypoint_type`` instead.